lib/css/parser/plugins/modules/selectors.js

```diff
@@ -0,0 +1,249 @@
+				}
+
+				return [...acc, n];
+
+			case "operator":
+				if (isModifier(nodes[index + 1])) {
+					return [...acc, Object.assign({}, n, { after: "" })];
+				}
+
+				return [...acc, n];
+
+			case "pseudo-class":
+				if (isModifier(n)) {
+					if (inside) {
+						throw Error(
+							`A :${n.name} is not allowed inside of a :${inside}(...)`
+						);
+					}
+
+					if (index !== 0 && !isSpacing(nodes[index - 1])) {
+						throw Error(`Missing whitespace before :${n.name}`);
+					}
+
+					if (index !== nodes.length - 1 && !isSpacing(nodes[index + 1])) {
+						throw Error(`Missing whitespace after :${n.name}`);
+					}
+					// set mode
+					mode = n.name;
+
+					return acc;
+				}
+
+				return [...acc, n];
+
+			case "nested-pseudo-class":
+				if (n.name === "local" || n.name === "global") {
+					if (inside) {
+						throw Error(
+							`A :${n.name}(...) is not allowed inside of a :${inside}(...)`
+						);
+					}
+					return [
+						...acc,
+						...localizeNode(n.nodes[0], {
+							mode: n.name,
+							inside: n.name,
+							getAlias
+						}).nodes
+					];
+				} else {
+					return [
+						...acc,
+						Object.assign({}, n, {
+							nodes: localizeNode(n.nodes[0], { mode, inside, getAlias }).nodes
+						})
+					];
+				}
+
+			case "id":
+			case "class":
+				if (mode === "local") {
+					return [...acc, Object.assign({}, n, { name: getAlias(n.name) })];
+				}
+
+				return [...acc, n];
+
+			default:
+				return [...acc, n];
+		}
+	}, []);
+
+	return Object.assign({}, node, { nodes: trimNodes(newNodes) });
+}
+
+const localizeSelectors = (selectors, mode, getAlias) => {
+	const node = Tokenizer.parse(selectors);
+
+	return Tokenizer.stringify(
+		Object.assign({}, node, {
+			nodes: node.nodes.map(n => localizeNode(n, { mode, getAlias }))
+		})
+	);
+};
+
+const walkRules = (css, callback) => {
+	css.walkRules(rule => {
+		if (rule.parent.type !== "atrule" || !/keyframes$/.test(rule.parent.name)) {
+			callback(rule);
+		}
+	});
+};
+
+const flatten = arr => arr.reduce((acc, item) => [...acc, ...item], []);
+
+const isRedeclared = (messages, name) => {
+	return messages.find(msg => msg.type === "icss-scoped" && msg.name === name);
+};
+
+const getValue = (messages, name) => {
+	return messages.find(msg => msg.type === "icss-value" && msg.value === name);
+};
+
+const getMessages = aliases => {
+	// TODO(michael-ciniawsky)
+	console.log(aliases);
+
+	Object.keys(aliases).map(name => ({
+		plugin,
+		type: "icss-scoped",
+		name,
+		value: aliases[name]
+	}));
+};
+
+const getComposed = (name, messages, root) => [
+	name,
+	...flatten(
+		messages
+			.filter(msg => msg.name === name && msg.value !== root)
+			.map(msg => getComposed(msg.value, messages, root))
+	)
+];
+
+const mapMessages = (messages, type) => {
+	return fromPairs(
+		messages.filter(msg => msg.type === type).map(msg => [msg.name, msg.value])
+	);
+};
+
+// const composeAliases = (aliases, messages) =>
```

Comment:
Should this still be here if it’s commented out?

Reply:
The whole `plugins` folder will be removed soon and upstreamed to `css-modules` instead, so the postcss plugins for `@import, url()` and CSS Modules will be separate packages

Reply:
No it shouldn't :)